plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_incident_events/task.py

```diff
@@ -0,0 +1,91 @@
+import insightconnect_plugin_runtime
+from .schema import (
+    MonitorIncidentEventsInput,
+    MonitorIncidentEventsOutput,
+    MonitorIncidentEventsState,
+    Input,
+    Output,
+    Component,
+    State,
+)
+
+# Custom imports below
+import json
+
+from ...util.util import Util
+from typing import List, Dict
+
+
+INITIAL_HISTORICAL_LOOK_BACK_MS = 1 * 60 * 60 * 1000
+DEFAULT_TIME_SORTING_FIELD = "modification_time"
+
+
+class MonitorIncidentEvents(insightconnect_plugin_runtime.Task):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_incident_events",
+            description=Component.DESCRIPTION,
+            input=MonitorIncidentEventsInput(),
+            output=MonitorIncidentEventsOutput(),
+            state=MonitorIncidentEventsState(),
+        )
+
+    @staticmethod
+    def get_request_filters(status: str, descriptions: List[str], incident_ids: List[str]) -> List[Dict]:
+        filters = []
+        if status is not None and len(status) > 0:
+            filters.append({"field": "status", "operator": "eq", "value": status})
+
+        if descriptions is not None and len(descriptions) > 0:
+            filters.append({"field": "description", "operator": "in", "value": descriptions})
+
+        if incident_ids is not None and len(incident_ids) > 0:
+            filters.append({"field": "incident_id_list", "operator": "in", "value": incident_ids})
+
+        return filters
+
+    @staticmethod
+    def remove_old_and_sort_incident_events(
+        incident_events: List[Dict], epoch_cutoff: int, time_sorting_field: str
+    ) -> List[Dict]:
+        # Discard events < epoch cutoff. This avoid dupes since the XDR API only accepts `>=` and not simply `>`
+        events_after_cutoff = [x for x in incident_events if x.get(time_sorting_field, -1) > epoch_cutoff]
+        events_after_cutoff.sort(key=lambda x: x.get(time_sorting_field, -1))
+        return events_after_cutoff
+
+    def run(self, params={}, state={}):
+        # Get all input variables
+        status_filter_value = params.get(Input.STATUS, None)
+        descriptions_filter_value = params.get(Input.DESCRIPTIONS, None)
+        incident_id_list_filter_value = params.get(Input.INCIDENT_ID_LIST, None)
```

Comment:
```suggestion
        status_filter_value = params.get(Input.STATUS)
        descriptions_filter_value = params.get(Input.DESCRIPTIONS)
        incident_id_list_filter_value = params.get(Input.INCIDENT_ID_LIST)
```
`None` is the default for `.get()` :)